### DISCUSSION TITLE
How to display Transit Land routes and stops on QGIS/ArcGIS?

### DISCUSSION THREAD

**Discussion by stmsim** [transitland/transitland#355] [Q&A] [answered]
I find visualizing and examining routes and stops on the web version of Transitland (transit.map/land) is quite handy for me, as an analyst in a transit agency. However, I'd love to do that on a QGIS or ArcGIS project. Is there any way to use my API key to connect to a WMS/WMTS, Vector Tiles, or something equivalent? 

![image](https://user-images.githubusercontent.com/86686508/160135090-cd6ef302-527f-4f2d-8fad-69fb0712794f.png)

**Comment by drewda** [comment, marked as answer]
We do not provide WMS or WFS at this time, although those are useful ideas for the future.

Transitland does provide MVT (Mapbox Vector Tile) tiles through the the [v2 Vector Tiles](https://www.transit.land/documentation/vector-tiles) API. If your GIS software can accept MVT format, then you can configure two layers to point to:

```
https://transit.land/api/v2/tiles/routes/tiles/{z}/{x}/{y}.pbf?apikey=xxx
https://transit.land/api/v2/tiles/stops/tiles/{z}/{x}/{y}.pbf?apikey=xxx
```

Or you can use the v2 REST API [stop](https://www.transit.land/documentation/rest-api/stops) and [route](https://www.transit.land/documentation/rest-api/routes) endpoints to generate GeoJSON format, which can then be imported into your GIS software.

**Reply by stmsim** [reply]
Thanks! This method works with the Vector Tiles connector in QGIS. It reflects very well what is displayed on Transit.land/map, even though it has some trouble representing the eastern part of the US:

![image](https://user-images.githubusercontent.com/86686508/160674771-eee4094f-8c3c-4a92-9cb3-f3842afcd641.png)
![image](https://user-images.githubusercontent.com/86686508/160674952-dd00a278-c9ed-4763-b59f-f9d6b49020e1.png)

**Reply by drewda** [reply]
@stmsim glad to hear the QGIS Vector Tiles connector works well.

That is a very strange geometry across Pennsylvania. We will have to look into that.

**Reply by drewda** [reply]
We now have addressed this particularly problematic route geometry in the eastern US. 

We also wrote up a blog post with an overview of the improvements, and also instructions for others who wish to use Transitland tiles in QGIS in the future: https://www.transit.land/news/2022/06/10/vector-map-tiles

**Comment by plempert** [comment]
![Screen Shot 2023-07-28 at 8 06 52 AM](https://github.com/transitland/transitland/assets/11591815/444b08ba-1b8e-4e77-acd0-e91771a6ce34)
@drewda I am seeing a similar issue with the route tiles, especially around the NYC area. Is there any way to fix this?

**Reply by drewda** [reply]
This is due to a change since 2022. For more information, see https://github.com/transitland/transitland/discussions/398#discussioncomment-6307373

**Reply by plempert** [reply]
Thanks, I will take a look!

**Reply by plempert** [reply]
For anyone with the same issue, I was able to resolve it in mapbox gl with the following code (see 'filter' in addLayer)

```
const map = new mapboxgl.Map({
container: 'map', // container ID
// Choose from Mapbox's core styles, or make your own style with Mapbox Studio
style: 'mapbox://styles/mapbox/streets-v12', // style URL
center: [-74.006, 40.712], // starting position [lng, lat]
zoom: 14, // starting zoom
});

map.on('load', function () {
    map.addSource('routes-source', {
        'type': 'vector',
    tiles: [`https://transit.land/api/v2/tiles/routes/tiles/{z}/{x}/{y}.pbf?apikey=${apikey}`],
    maxzoom: 14
    });
    
    map.addLayer({
        'id': 'routes-layer',
        'type': 'line',
        'source': 'routes-source',
        'source-layer': 'routes',
        'filter': ['<', 'geometry_max_segment_length', 5000]
    });
    
});
```

**Reply by drewda** [reply]
Thanks for sharing :)
